synoptic/src/synoptic/tests/units/JsonExporterTests.java

```diff
@@ -0,0 +1,510 @@
+package synoptic.tests.units;
+
+import static org.junit.Assert.assertTrue;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.junit.Before;
+import org.junit.Test;
+
+import synoptic.main.AbstractMain;
+import synoptic.main.PerfumeMain;
+import synoptic.main.options.PerfumeOptions;
+import synoptic.model.PartitionGraph;
+import synoptic.model.export.DotExportFormatter;
+import synoptic.model.export.JsonExporter;
+import synoptic.model.export.types.EvBasedGraph;
+
+public class JsonExporterTests {
+
+    private PerfumeMain main = null;
+    private PartitionGraph pGraph = null;
+    private EvBasedGraph evGraph = null;
+    private Object expectedJSON = null;
+    private JSONObject expectedJSONObject = null;
+    private JSONParser parser = null;
+    private String dispJString = null;
+
+    @Before
+    public void setUp() throws Exception {
+        // Set up Perfume options
+        PerfumeOptions perfOpts = new PerfumeOptions();
+        perfOpts.regExps = Arrays.asList(
+                new String[] { "(?<ip>.+), (?<TYPE>.+), (?<DTIME>.+)" });
+        perfOpts.partitionRegExp = "\\k<ip>";
+        perfOpts.logFilenames.add(
+                "./traces/abstract/perfume-survey/browser-caching-traces.txt");
```

Comment:
As we discussed over email, different methods of running these unit tests result in different behaviors when using relative paths like this one (the current directory when running the tests is either $repo/synoptic/synoptic or $repo/synoptic/synoptic/bin). Therefore, this test will fail when run in some ways and pass in others.

Add a static method somewhere (perhaps SynopticTest) that resolves paths for files used by unit tests. Make the method something like `getTestPath(String path)` that takes a relative path like this--but probably omit the `./`--and checks if the file as written works. If not, it checks if `../$pathAsWritten` works. If neither work, it throws an exception. Call that new method from here to get the path for this txt file.